src/Buildalyzer/ProjectAnalyzer.cs

```diff
@@ -186,16 +202,66 @@ public void SetGlobalProperty(string key, string value)
             {
                 throw new InvalidOperationException("Can not change global properties once project has been loaded");
             }
-            _buildEnvironment.GlobalProperties[key] = value;
+            _globalProperties[key] = value;
         }
 
-        public bool RemoveGlobalProperty(string key)
+        public void RemoveGlobalProperty(string key)
         {
             if (_project != null)
             {
                 throw new InvalidOperationException("Can not change global properties once project has been loaded");
             }
-            return _buildEnvironment.GlobalProperties.Remove(key);
+
+            // Nulls are removed before passing to MSBuild and can be used to ignore values in lower-precedence collections
+            _globalProperties[key] = null;
+        }
+
+        public void SetEnvironmentVariable(string key, string value)
+        {
+            if (_project != null)
+            {
+                throw new InvalidOperationException("Can not change environment variables once project has been loaded");
+            }
+            _environmentVariables[key] = value;
+        }
+
+        // Note the order of precedence (from least to most)

```

Comment:
Basically the `BuildEnvironment` variables are applied first, then the `AnalyzerManager` ones are applied (which may override the `BuildEnvironment` ones), and finally the `ProjectAnalyzer` ones are applies, which can override the other two.